services/galley/src/Galley/Intra/Client.hs

```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Galley.Intra.Client
+    ( lookupClients
+    ) where
+
+import Bilge hiding (options, getHeader, statusCode)
+import Bilge.RPC
+import Bilge.Retry
+import Galley.App
+import Galley.Options
+import Galley.Types (UserClients)
+import Control.Lens (view)
+import Control.Retry
+import Data.ByteString (ByteString)
+import Data.ByteString.Char8 (intercalate)
+import Data.ByteString.Conversion
+import Data.Id
+import Data.Misc (portNumber)
+import Data.Text.Encoding (encodeUtf8)
+import Data.Word (Word16)
+import Network.HTTP.Types.Method
+import Network.HTTP.Types.Status
+import Network.Wai.Utilities.Error
+import Util.Options
+
+import qualified Data.ByteString.Lazy as LB
+import qualified Data.Text.Lazy as LT
+
+lookupClients :: [UserId] -> Galley UserClients
+lookupClients uids = do
+    (h, p) <- brigReq
+    r <- call "brig"
+        $ method GET . host h . port p
+        . path "/i/clients"
+        . queryItem "ids" users
```

Comment:
As discussed, there can be limits imposed on the size of the query parameter. The broadcast endpoint could be looking up clients of up to 1127 users at once, resulting in a length of the query string of 41703 characters. Please add a test for this. (it might be useful to do this lookup in batches regardless of the outcome of a length test)

Reply:
I am not sure adding a test for that case makes sense. It is fairly easy to convince yourself, as you just did, that this won't work in general.

We could split up the request, but I think this leads to some other unnecessary complications (e.g. in case of failed requests etc). I think it would be better to make the HTTP call work in general, by using POST with e.g. a JSON payload. If we are thinking in terms of performance I would rather parallelize the DB calls on the Brig side.